### DISCUSSION TITLE
Should large objects be used in x-data?

### DISCUSSION THREAD

**Discussion by tmrkp** [alpinejs/alpine#3516] [1. Help]
I've been using Swiper together with Alpine in a website like this:

```html
<div x-data="{ swiper: null }">
  <swiper-container  x-init="swiper = $el.swiper">
    <swiper-slide>...</swiper-slide>
    <swiper-slide>...</swiper-slide>
    <swiper-slide>...</swiper-slide>
  </swiper-container>
  ...
</div>
```

This makes it very convenient to work with Swiper, because the swiper object is now reactive and I can use properties like `swiper.activeIndex` without working with events. But I was wondering if this is accually good practice. I'm not sure what's exactly happeing behind the scenes in the reactivity engine or if this could lead to problems. Does anyone have insights or thoughts on this?

**Comment by ekwoka** [comment]
It shouldn't be much of an issue. The bigger issue is using swiper at all, when you can do everything it does pretty dang easily in Alpine directly.

Things inside of swiper themselves won't all be reactive, like if it communicates with itself in a way that doesn't use `this` without moving down the tree of the target. I'd have to double check, but the reactivity engine is lazy. It doesn't wrap things proactively, but only if you actually are accessing that chunk of data.

**Reply by tmrkp** [reply]
Okay sound good! And now that I think of it using Swiper might accually be a little overkill, since I'm only using it to fade some images.

My initial thought was, since the entire swiper object gets wrapped in a Proxy then whenever something in this object changes due to swiper doing it's thing. This might cause Alpine to react to that and do something even though nothing thats 'concerning' Alpine is happening.
